cmd/coordinator.go

```diff
@@ -264,6 +272,251 @@ var estimateDepositsSweepFeeCommandDescription = "Estimates the satoshi " +
+		}
+
+		transactionHashFlag, err := cmd.Flags().GetString(transactionHashFlagName)
+		if err != nil {
+			return fmt.Errorf("failed to find transaction hash flag: %v", err)
+		}
+
+		transactionHash, err := bitcoin.NewHashFromString(
+			transactionHashFlag,
+			bitcoin.ReversedByteOrder,
+		)
+		if err != nil {
+			return fmt.Errorf("failed to parse transaction hash flag: %v", err)
+		}
+
+		// Allow the caller to request a specific number of confirmations.
+		// The Bridge calculates the required difficulty of a chain of block
+		// headers by multiplying the difficulty of the first block header by
+		// the difficulty factor. If the block headers happen to span the
+		// Bitcoin epoch difficulty change and there is a drop of difficulty
+		// between the epochs, the sum of difficulties from the headers chain
+		// may be too low. Allowing the caller to specify a greater number of
+		// confirmations will ensure the transaction can be proven.
+		requiredConfirmations, err := cmd.Flags().GetUint(confirmationsFlagName)
+		if err != nil {
+			return fmt.Errorf("failed to get confirmations flag: %v", err)
+		}
+
+		if requiredConfirmations == 0 {
+			// The caller did not specify the number of confirmations. Use the
+			// default  value stored onchain.
+			txProofDifficulty, err := tbtcChain.TxProofDifficultyFactor()
+			if err != nil {
+				return fmt.Errorf(
+					"failed to get transaction proof difficulty factor: %v",
+					err,
+				)
+			}
+
+			requiredConfirmations = uint(txProofDifficulty.Int64())
+		}
+
+		logger.Infof(
+			"Preparing deposit sweep proof for transaction %s with %v "+
+				"confirmations",
+			transactionHashFlag,
+			requiredConfirmations,
+		)
+
+		transaction, proof, err := bitcoin.AssembleSpvProof(
+			transactionHash,
+			requiredConfirmations,
+			btcChain,
+		)
+		if err != nil {
+			return fmt.Errorf("failed to assemble transaction spv proof: %v", err)
+		}
+
+		mainUTXO, vault, err := parseTransactionInputs(
+			btcChain,
+			*tbtcChain,
+			transaction,
+		)
+		if err != nil {
+			return fmt.Errorf("error while parsing transaction inputs: %v", err)
+		}
+
+		if err := tbtcChain.SubmitDepositSweepProofWithReimbursement(
+			transaction,
+			proof,
+			mainUTXO,
+			vault,
+		); err != nil {
+			return fmt.Errorf(
+				"failed to submit deposit sweep proof with reimbursement: %v",
+				err,
+			)
+		}
+
+		logger.Infof(
+			"Successfully submitted deposit sweep proof for transaction: %s",
+			transactionHashFlag,
+		)
+
+		return nil
+	},
+}
+
+// convertVaultAddress converts the vault's address from the chain's address
+// to the common address. If the chain's address is nil, the returned value
+// is zero-filled.
+func convertVaultAddress(vault *chain.Address) common.Address {
+	if vault == nil {
+		return common.Address{}
+	}
+
+	return common.HexToAddress(string(*vault))
+}
+
+// parseTransactionInputs parses the transaction's inputs and returns the main
+// UTXO and the vault.
+func parseTransactionInputs(
```

Comment:
Done in 37c320fd126ea9cb8b408f2e64333c0c707eedc2.
I moved most of the code into `pkg/maintainer/spv`, inside a function like this:
```
SubmitDepositSweepProof(
	transactionHash bitcoin.Hash,
	requiredConfirmations uint,
	btcChain bitcoin.Chain,
	tbtcChain tbtc.Chain,
) error
```
However, I left some code, inside the `cmd` package: parsing of the command's parameters and creating handles to BTC and TBTC chains. 
The reason for this is that the chains can now be injected into the function.
Also the TBTC chain is needed in `cmd` to properly set the value of `requiredConfirmations` (it can be either set by the user or read from the TBTC chain).